sympy/geometry/hyperbola.py

```diff
@@ -0,0 +1,703 @@
+"""Hyperbolic geometrical entities.
+
+Contains
+* Hyperbola
+
+"""
+
+from sympy.core import S, sympify
+from sympy.core.evalf import N
+from sympy.core.logic import fuzzy_bool
+from sympy.core.numbers import Rational
+from sympy.core.symbol import Dummy, _symbol, symbols
+from sympy.simplify import simplify, trigsimp
+from sympy.functions.elementary.miscellaneous import sqrt, Max
+from sympy.functions.elementary.trigonometric import sec, tan
+from .entity import GeometryEntity, GeometrySet
+from .exceptions import GeometryError
+from .ellipse import Circle
+from .line import Segment
+from .point import Point
+from sympy.utilities.misc import filldedent
+
+
+class Hyperbola(GeometrySet):
+    """A hyperbolic GeometryEntity.
+
+    Parameters
+    ==========
+
+    center : Point, optional
+        Default value is Point(0, 0)
+    hradius : number or SymPy expression, optional
+    vradius : number or SymPy expression, optional
+    eccentricity : number or SymPy expression, optional
+        Two of `hradius`, `vradius` and `eccentricity` must be supplied to
+        create an Hyperbola. The third is derived from the two supplied.
+
+    Attributes
+    ==========
+
+    center
+    hradius
+    vradius
+    eccentricity
+    focus_distance
+    foci
+
+    Raises
+    ======
+
+    GeometryError
+        When `hradius`, `vradius` and `eccentricity` are incorrectly supplied
+        as parameters.
+    TypeError
+        When `center` is not a Point.
+
+    See Also
+    ========
+
+    Ellipse
+
+    Notes
+    -----
+    Constructed from a center and two radii, the first being the horizontal
+    radius (along the x-axis) and the second being the vertical radius (along
+    the y-axis).
+
+    When symbolic value for hradius and vradius are used, any calculation that
+    refers to the foci or the major or minor axis will assume that the hyperbola
+    has its major radius on the x-axis. If this is not true then a manual
+    rotation is necessary.
+
+    Examples
+    ========
+
+    >>> from sympy import Hyperbola, Point
+    >>> h1 = Hyperbola(Point(0, 0), 5, 3)
+    >>> h1.hradius, h1.vradius
+    (5, 3)
+    >>> h2 = Hyperbola(Point(3, 1), hradius=3, eccentricity= 2)
+    >>> h2
+    Hyperbola(Point2D(3, 1), 3, 3*sqrt(3))
+
+    """
+
+    def __contains__(self, o):
+        if isinstance(o, Point):
+            x = Dummy('x', real=True)
+            y = Dummy('y', real=True)
+
+            res = self.equation(x, y).subs({x: o.x, y: o.y})
+            return trigsimp(simplify(res)) is S.Zero
```

Comment:
This should probably be `trigsimp(simplify(res)).is_zero` unless `contains` cannot return None. Otherwise if it must be T/F then an Undecidable (sp?) error should be raised if it doesn't resolve.